src/TwitchRecover.Core/WebsiteRetrieval.java

```diff
@@ -172,40 +129,70 @@ private static String getJSON(String url) throws IOException {
      * @return String[4]    String array containing the 4 principal values (streamer's name, stream ID,
      * timestamp of the start of the stream and the duration) in that respective order.
-     * @throws IOException
      */
     private static String[] getSCData(String url) throws IOException {
         String[] results = new String[4];     //0: streamer's name; 1: Stream ID; 2: Timestamp; 3: Duration.
-        String userID;
-        double duration = 0.0;
         //Retrieve initial values:
-        String pattern = "streamscharts\\.com\\/twitch\\/channels\\/([a-zA-Z0-9_-]*)\\/streams\\/(\\d*)";
+        String pattern = "streamscharts\\.com/channels/([a-zA-Z0-9_-]*)/streams/(\\d*)";
         Pattern r = Pattern.compile(pattern);
         Matcher m = r.matcher(url);
         if(m.find()) {
             results[0] = m.group(1);
             results[1] = m.group(2);
         }
 
-        //Retrieve user ID:
-        String idJSON = getJSON("https://api.twitch.tv/v5/users/?login=" + results[0] + "&client_id=ohroxg880bxrq1izlrinohrz3k4vy6");
-        JSONObject joID = new JSONObject(idJSON);
-        JSONArray users = joID.getJSONArray("users");
-        JSONObject user = users.getJSONObject(0);
-        userID = user.getString("_id");
-
-        //Retrieve stream values:
-        String dataJSON = getJSON("https://alla.streamscharts.com/api/free/streaming/platforms/1/channels/" + userID + "/streams/" + results[1] + "/statuses");
-        JSONObject joD = new JSONObject(dataJSON);
-        JSONArray items = joD.getJSONArray("items");
-        for(int i = 0; i < items.length(); i++) {
-            JSONObject item = items.getJSONObject(i);
-            if(i == 0) {
-                results[2] = item.getString("stream_created_at");
+        Document doc = Jsoup.connect(url).get();
+        Element airTimeDom = doc.select("span.mx-2").first();
+        Element startDateDom = doc.select("time.ml-2").first();
+
+
+        // START TIME
+
+        SimpleDateFormat streamsChartsFormatString = new SimpleDateFormat("dd MMM yyyy, HH:mm zzz", Locale.ENGLISH);
+        SimpleDateFormat formatStringForCompute = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss zzz", Locale.ENGLISH);
+        Date startDate = null;
+        try {
+            if (startDateDom != null)
+            {
+                startDate = streamsChartsFormatString.parse(startDateDom.text() + " GMT+00:00");
             }
-            duration += item.getDouble("air_time");
+        } catch (ParseException e) {
+            e.printStackTrace();
         }
-        results[3] = String.valueOf(duration * 60);
+
+        results[2] = formatStringForCompute.format(startDate);
+
+        if (airTimeDom != null) {
+            results[3] = String.valueOf(airTimeFromString(airTimeDom.text().replace(",", "").replace(" ", "")) * 60);
+        }
+
         return results;
     }
-}
\ No newline at end of file
+
+    private static Integer airTimeFromString(String airTimeString) {
```

Comment:
I think it is better to use multiple groups.
`([0-9]+h)([0-9]+m)`

Reply:
ah streams chart trims (\d+h) if the hour is zero

Reply:
You are right, that looks cleaner. Resolved with commit [bb242e7](https://github.com/TwitchRecover/TwitchRecover/pull/195/commits/bb242e793b73720fb0921393e9e0d2712533c33b)

I changed your regex a bit because it had problems with a stream with less than 1h airtime